src/controllers/channel.controller.js

```diff
@@ -0,0 +1,129 @@
+import { getConnection } from '../database.js'
+import Joi from 'joi'
+const dbTable = 'Channel'
+
+const getChannels = async (req, res) => {
+	try {
+		const connection = await getConnection()
+		const dbQuery = `SELECT * FROM ${dbTable}`
+		const result = await connection.query(dbQuery)
+		res.json(result).status(200)
+	} catch (error) {
+		res.status(500)
+		res.send(error.message)
+	}
+}
+
+const getChannel = async (req, res) => {
+	try {
+		const connection = await getConnection()
+		const dbQuery = `SELECT * FROM ${dbTable} WHERE id = ?`
+		const queryValues = [parseInt(req.params.channelId)]
+		const result = await connection.query(dbQuery, queryValues)
+
+		if (!result[0]) {
+			res.json({ error: 'The channel you are looking for does not exist!' }).status(204)
+			return
+		}
+
+		res.json(result).status(200)
+	} catch (error) {
+		res.status(500)
+		res.json({ error: error.message })
+	}
+}
+
+const addChannel = async (req, res) => {
+	try {
+		const schema = Joi.object({
+			authorId: Joi.number().integer().required(),
```

Comment:
I think the best way to get the `author_id` is to extract it from the auth token's payload, but let's stick to this way for now. coz we will discuss how we gonna authenticate the user in the next scrum.